### ISSUE TITLE
README

### ISSUE THREAD

**Issue by MWesselkamp** [therealniklasmoser/physics_guided_nn#8]
What to discuss the next zoom:

- [x] Folder structure: when I pull and push, there should be no changes in folder structure. Do you also save your results and data to results and data folder? In code folder or outside?

**Comment by MWesselkamp** [comment]
AS, HP and LR choices not only based on min_mean_valloss but also min_sd_valloss. Accuracy and Precision.
**Now selected as**: (mean_val_loss^2 + mean_val_std^2)/2

**Comment by MWesselkamp** [comment]
Residual network (evalres.py)
Only GPP as input - why map all three to GPP obs?
**Changed**: Standardize training and test set together.

**Comment by MWesselkamp** [comment]
Suggestion: Use year 2004, 2005, 2007 for NAS. Throw away afterwards. Use 2008-2012 for evaluation.
Means, calibrating Preles at the same data. **Calibration only on GPP will be biased?!**
Peltoniemi calibrated the 13 parameters simulateously at GPP and E.
Way to do this: Loglik(GPP) + Loglik(ET) with standardized variables. Weighted? Why assuming gaussian likelihood.

**At the moment only calibrated at GPP with gaussian density.**

- [ ] Decision on Calibration.

**Comment by MWesselkamp** [comment]
**Why not use the same architecture for the naive, the parallel and the regularized network?**

Two reasons why we should:
1. IF NOT, we have to include PRELES estimates for GPP, ET and SW. When we use the two years 2004/2005 only for the NAS, we would also have to use only those to years to calibrate PRELES here, or use default parameter values (but why would we do that?). PRELES BC delivers very bad results for only those two years.
2. Much better comparable to the naive network when we use the same architecture.

**At the moment using default params for PRELES predictions on 2004 and 2005**
Available as hyytialaNAS.csv

**Comment by MWesselkamp** [comment]
- [x] Debug pretraining: Simulations correct?
- [x] Debug pretraining: use a small source domain
- [ ] Pretraining: Samples from all 13 parameters calibrated?
- [ ] (Quantify similarity between source and target domain?)
- [ ] (AS at source AND target domain?)
- [x] Enable data_use for model saving in pretraining.
- [x] Multisite Calibration of Preles
- [x] Updata multisite data set.
- [x] Update and upload train_/test_hyyt.csv 
- [x] Adjust NAS loaded years in utils.py: 2004/5
- [x] Adjust train-test split in LR-finetuning: Training in CV. Split 2004/5 in 5.
- [x] Check fapar: remove 2007 completely. 
- [x] Select AS&HP&LR by ind_mini: Adjust in LR-finetuning and evaluation.

**Comment by MWesselkamp** [comment]
**New datasets for utils.loaddata()**

"NAS": hyytialaNAS.csv
"validation": hyytialaF.csv
"exp2": allsitesF.csv

Generated in file *code/FitPRELES.R*

**Comment by MWesselkamp** [comment]
HP & A search as well as LR finetuning are not done in a CV setting.
If we use only two years here, we have to do this in a cross-validation:

--> select HP & A by the mean(val_losses_mean) and mean(val_losses_sd) in 5 fold CV.
OR by mean(val_losses_mean) and sd(val_losses_mean)?! 

- [ ] Implement CV for hp=True and in mlpft.
